editor/js/Sidebar.Material.js

```diff
@@ -493,6 +511,15 @@ Sidebar.Material = function ( editor ) {
 		var geometry = object.geometry;
 		var material = object.material;
 
+		var previousSelectedSlot = currentMaterialSlot;
+
+		currentMaterialSlot = materialSlotSelect.getValue() | 0;
```

Comment:
I use it because it's faster, and doesn't fail on bad input.. I normally don't do defensive programming like that if I am writing both endpoints, but with data coming in from a UI layer or possibly an editable widget, I did a little extra checking. I think parseInt returns NaN on "" or other malformed input. But I'll replace the | 0 with parseInts if you prefer.